### PR TITLE
Click event for an image component

### PR DESCRIPTION
**Pull request by moliata**
Hello there,
this is a pull request which adds a click event for an image component. I have:

- [x] Added a click event to an image component
- [x] Updated a version of an image component
- [x] Upgraded an image component (form upgrader side)
- [x] Upgraded an image component (versioning side)
- [x] Added an event information in a documentation

Best regards, Ben (@moliata on GitHub)

### CODE REVIEW

**Comment by ewpatton** [comment]
@moliata What is the benefit of this versus using Button?

**Comment by moliata** [comment]
While button is used for being a clickable-only object, image component is used for displaying images and so, this component has some cool features like: RotationAngle, ScalePictureToFit, Animations. Making a clickable animated picture is way better than trying to somehow create such button. Personally I am using RotationAngle property together with image's click event.
P.S: We would also can update Hello Purr example project.

**Comment by halatmit** [comment]
@moilata

This sounds like useful enhancement.   Have you tried implementing it
yourself?  I'm guessing it that should be possible to add a few lines to
Image.java
similar to the the lines that implement OnClick for buttons in
ButtonBase.java.     Do you want to try doing that
and submit a pull request?  (Note that this would be a component upgrade.)



== Hal

Hal Abelson
hal@mit.edu
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.

File: appinventor/docs/reference/components/userinterface.html [hunk 1]
```diff
@@ -667,7 +667,10 @@ <h3>Properties</h3>
                  </dl>
 
                  <h3>Events</h3>
-                 none
+                 <dl>
+		   <dt><code>Click()</code></dt>
+		   <dd>User tapped and released an image</dd>
```
Comment (ewpatton) [inline appinventor/docs/reference/components/userinterface.html]: You may want to make this text match the description given in the `@SimpleEvent` annotation in the sources.

File: appinventor/docs/reference/components/userinterface.html [hunk 2]
```diff
@@ -667,7 +667,10 @@ <h3>Properties</h3>
...
+		 </dl>
```
Comment (ewpatton) [inline appinventor/docs/reference/components/userinterface.html]: Minor whitespace change--this is one less space in than the opening `<dl>` tag.

File: appinventor/components/src/com/google/appinventor/components/common/YaVersion.java [hunk 3]
```diff
@@ -419,8 +419,10 @@ private YaVersion() {
   // For YOUNG_ANDROID_VERSION 166:
   // - MAP_COMPONENT_VERSION was incremented to 3
   // - FEATURE_COLLECTION_COMPONENT_VERSION was incremented to 2
+  // For YOUNG_ANDROID_VERSION 167:
+  // - IMAGE_COMPONENT_VERSION was incremented to 4
 
-  public static final int YOUNG_ANDROID_VERSION = 166;
+  public static final int YOUNG_ANDROID_VERSION = 167;
```
Comment (ewpatton) [inline appinventor/components/src/com/google/appinventor/components/common/YaVersion.java]: This isn't a hard policy yet, but we are thinking that `YOUNG_ANDROID_VERSION` should only be updated when we officially publish to a server, similarly to how we handle the change to the Companion version. This is okay for now (just a heads up).

**Comment by moliata** [comment]
@halatmit this is a pull request, not an issue 😄 
@ewpatton fixing that currently.

**Comment by moliata** [comment]
@ewpatton did it, please review this again.

**Comment by moliata** [comment]
@ewpatton sure, thanks!

**Comment by NmdOfficial** [comment]
Hello guy's.
Don't forget that if you do it like this the image component will be then always clickable no matter if the developer wants that or not.
A better method will be a designer property to enable or disable the click option.

Best regards,
Mika from Makeroid

**Comment by moliata** [comment]
@NmdOfficial hello Mika. I think that if user wants to catch image component's clicks he can use Click event and if doesn't want that, he can just not use it.

**Comment by NmdOfficial** [comment]
No matter if the user use the event or not he will hear a click sound.

**Comment by moliata** [comment]
@jisqyv what errors are you getting?

**Comment by ewpatton** [comment]
I think you forgot to import SimpleEvent.

**Comment by moliata** [comment]
@ewpatton thanks, I'll add it in the mean time as well as a property.

**Comment by moliata** [comment]
@ewpatton sure, a lot of studies lately, so during the Christmas holiday, I'll try to rearrange my time and do a couple of changes based on the requests, thanks!

**Comment by ewpatton** [comment]
@moliata We're planning to do a release soon. Do you want to aim to get this in before then or should I move it to the next companion release?

**Comment by AppInventorWorkerBee** [comment]
Can one of the admins verify this patch?

**Comment by moliata** [comment]
@ewpatton Thanks for reminding, been busy with International Baccalaureate studies and projects recently, I'll push make an update by the end of this weekend.

**Comment by moliata** [comment]
Currently working on updates, thanks for patience.

**Comment by ewpatton** [comment]
@moliata Any thoughts on when you will have a chance to work on this?

**Comment by moliata** [comment]
Check #2005 by @conorshipp as it superseded this PR.

**Pull request closed without merging**